src/common/shared.ts

```diff
@@ -139,10 +148,16 @@ export async function getCodeForFailedResult(
   if (!failedJourneyStep) return "";
 
   const failedStep = steps.find(
-    step => step.length > 0 && step[0].title === failedJourneyStep.name
+    step =>
+      step.actions.length > 0 &&
+      step.actions[0].title === failedJourneyStep.name
```

Comment:
I'm going to leave these code-level comments unresolved so that we have them, but I'm only looking for UI-level changes to be reviewed at the moment, with the goal of identifying visual bugs and problems with the UX that need resolution.

I've updated the main description to reflect this fact as well. This way we avoid reviewing code that is shared by #171, which this PR is built on top of.